### PR TITLE
InterApp - Launch App

### PR DESCRIPTION
**Pull request by AndrewEQ**
- added support for launching another app in android via its app package name, iOS still uses custom schemes

### CODE REVIEW

File: Source/Fuse.Launcher.InterApp/InterApp/JS.uno [hunk 1]
```diff
@@ -211,5 +212,29 @@ namespace Fuse.Reactive.FuseJS
...
+				http://<website address>
+				https://<website address>
+				tel:<phone number>
+				sms:<phone number>
+			
+			More information on supported URI schemes on iOS(https://developer.apple.com/library/content/featuredarticles/iPhoneURLScheme_Reference/Introduction/Introduction.html).
+		*/
+		public static object LaunchApp(Scripting.Context context, object[] args)
+		{
+
```
Comment (mortend) [inline Source/Fuse.Launcher.InterApp/InterApp/JS.uno]: Nit: Superfluous newline.

File: Source/Fuse.Launcher.InterApp/InterApp/InterAppTrigger.uno [hunk 2]
```diff
@@ -49,4 +49,13 @@ namespace Fuse.Triggers.Actions
 			Fuse.LauncherImpl.InterAppLauncher.LaunchUri(new Uri(this.Uri));
 		}
 	}
+	public class LaunchApp : TriggerAction
```
Comment (mortend) [inline Source/Fuse.Launcher.InterApp/InterApp/InterAppTrigger.uno]: Nit: Missing empty line between classes.

File: Source/Fuse.Android/AndroidInterop.uno [hunk 3]
```diff
@@ -29,6 +29,24 @@ namespace Fuse.Android.Bindings
...
+		public static extern(Android) Java.Object LaunchApp(string action, string applicationId)
+		@{
+			Intent pendingIntent = null;
+			Activity a = com.fuse.Activity.getRootActivity();
+
+			try
+			{
+				pendingIntent = a.getPackageManager().getLaunchIntentForPackage(applicationId);
+				a.startActivity(pendingIntent);
+			} catch (Exception e) {
```
Comment (mortend) [inline Source/Fuse.Android/AndroidInterop.uno]: Consistency: Either put all braces on their own lines, or move the first brace to become `try {`.

File: Source/Fuse.Launcher.InterApp/InterApp/InterAppLauncher.uno [hunk 4]
```diff
@@ -36,5 +36,22 @@ namespace Fuse.LauncherImpl
...
+
+		public static void LaunchApp(string uri)
+		{
+			if defined(Android)
+			{
+				Permissions.Request(Permissions.Android.INTERNET);
+				try {
+					AndroidDeviceInterop.LaunchApp(_actionView, uri);
+				} catch (Exception e) {
+					// ignore this error. Keeps behaviour in line with iOS
```
Comment (mortend) [inline Source/Fuse.Launcher.InterApp/InterApp/InterAppLauncher.uno]: Maybe we should report the exception to logcat at least?

File: Source/Fuse.Launcher.InterApp/InterApp/InterAppLauncher.uno [hunk 5]
```diff
@@ -36,5 +36,22 @@ namespace Fuse.LauncherImpl
 				iOSDeviceInterop.LaunchUriiOS(uri.AbsoluteUri);
 			}
 		}
+
+		public static void LaunchApp(string uri)
+		{
+			if defined(Android)
+			{
+				Permissions.Request(Permissions.Android.INTERNET);
```
Comment (mortend) [inline Source/Fuse.Launcher.InterApp/InterApp/InterAppLauncher.uno]: Why do we need `Permissions.Android.INTERNET`?

**Comment by mortend** [comment]
AppVeyor build failed because it failed to download Gradle.
```
Downloading https://services.gradle.org/distributions/gradle-5.6-all.zip
Exception in thread "main" java.net.ConnectException: Connection timed out: connect
```

This will probably fix itself if you `git push` one more time.

**Comment by AndrewEQ** [comment]
Yeah, bad habits copied from previous writer(s), sorted now.

**Pull request merged**